knotx-core/src/main/java/com/cognifide/knotx/placeholder/UriInfo.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Knot.x - Reactive microservice assembler
+ *
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+package com.cognifide.knotx.placeholder;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.collect.Maps;
+
+public class UriInfo {
+
+    private static final Pattern URI_PATTERN = Pattern
+            .compile("([^\\.\\?#]+)(\\.([^\\/\\?#]+)\\.)?(\\.?([^\\/\\?#]+))?([^\\?#]+)?((\\?|#).+)?");
+
+    private static final String PATH = "path";
+
+    private static final String PATHPART = "pathpart";
+
+    private static final String SELECTORSTRING = "selectorstring";
+
+    private static final String SELECTOR = "selector";
+
+    private static final String EXTENSION = "extension";
+
+    private static final String SUFFIX = "suffix";
+
+    private Map<String, String> values = Maps.newHashMap();
+
+    public UriInfo(String uri) {
```

Comment:
This UriInfo class should be request scope bean. Then all RequestPlaceholder's should be prototype beans with two parameters.

```java
@Bean
@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
public RequestPlaceholder requestPlaceholder(String placeholder) {
	return new RequestPlaceholder(placeholder);
}
```
Then you can use it as:
```java
applicationContext.getBean(RequestPlaceholder.class, placeholder);
```
In RequestPlaceholder class you can inject Uriinfo (which is request scope). Then in RequestPlaceholderResolver you will not keep any additional references. In that way you can simply add additional Placeholder's implementations.